docker_explorer/de.py

```diff
@@ -164,13 +166,48 @@ def ParseArguments(self):
     return opts
 
   def ParseOptions(self, options):
-    """Parses the command line options.
+    """Parses the command line options."""
+    self.docker_directory = os.path.abspath(options.docker_directory)
 
+  def GetContainer(self, container_id):
+    """Returns a Container object given a container_id.
+
+    Args:
+      container_id (str): the ID of the container.
     Returns:
-      Namespace: the populated namespace.
+      Container: the container's info.
     """
+    return container.Container(
+        self.docker_directory, container_id, docker_version=self.docker_version)
 
-    self.docker_directory = os.path.abspath(options.docker_directory)
+  def GetAllContainers(self):
+    """Gets a list containing information about all containers.
+
+    Returns:
+      list(Container): the list of Container objects.
+    """
+    container_ids_list = os.listdir(self.containers_directory)
+    if not container_ids_list:
+      print('Couldn\'t find any container configuration file (\'{0:s}\'). '
+            'Make sure the docker repository ({1:s}) is correct. '
+            'If it is correct, you might want to run this script'
+            ' with higher privileges.').format(
+                self.container_config_filename, self.docker_directory)
+    return [self.GetContainer(cid) for cid in container_ids_list]
+
+  def GetContainersList(self, only_running=False):
+    """Returns a list of container ids which were running, sorted by start date.
```

Comment:
Doesn't this return Container objects rather than container ids?

Reply:
fixed